RFC-0011-mixed-case-identifier-support.md

```diff
@@ -0,0 +1,234 @@
+  - Passing identifiers to connectors when creating new entities.
+  - Processing and displaying entity names retrieved from connectors, including column resolution and metadata introspection commands like SHOW and DESCRIBE.
+
+## Proposed Implementation
+
+#### Core Changes
+
+* In the presto-spi, add new API to pass original identifier (Schema, table and column names)
+* Introduce new API in Metadata for preserving lower case identifier by default to preserve backward compatibility
+* Introduce new Connector specific API in ConnectorMetadata
+
+Metadata.java
+
+```java
+    String normalizeIdentifier(Session session, String catalogName, String identifier, boolean delimited);
+```
+
+MetadataManager.java
+
+```java
+    @Override
+    public String normalizeIdentifier(Session session, String catalogName, String identifier, boolean delimited)
+    {
+        Optional<CatalogMetadata> catalogMetadata = getOptionalCatalogMetadata(session, transactionManager, catalogName);
+        if (catalogMetadata.isPresent()) {
+            ConnectorId connectorId = catalogMetadata.get().getConnectorId();
+            ConnectorMetadata metadata = catalogMetadata.get().getMetadataFor(connectorId);
+            return metadata.normalizeIdentifier(session.toConnectorSession(connectorId), identifier, delimited);
+        }
+        return identifier.toLowerCase(ENGLISH);
+    }
+```
+
+ConnectorMetadata.java
+```java
+
+    /**
+     * Normalize the provided SQL identifier according to connector-specific rules
+    */
+    default String normalizeIdentifier(ConnectorSession session, String identifier, boolean delimited)
+    {
+        return identifier.toLowerCase(ENGLISH);
+    }
+```
+
+JDBC Connector specific implementation
+
+JdbcMetadata.java
+
+```java
+    @Override
+    public String normalizeIdentifier(ConnectorSession session, String identifier, boolean delimited)
+    {
+        return jdbcClient.normalizeIdentifier(session, identifier, delimited);
+    }
+```
+
+JdbcClient.java
+```java
+    String normalizeIdentifier(ConnectorSession session, String identifier, boolean delimited);
+```
+
+BaseJdbcClient.java
+```java
+    @Override
+    public String normalizeIdentifier(ConnectorSession session, String identifier, boolean delimited)
+    {
+        return identifier.toLowerCase(ENGLISH);
+    }
+```
+
+Example - Connector specific implementation -
+MySqlClient.java
+
+```java
+    @Override
+    public String normalizeIdentifier(ConnectorSession session, String identifier, boolean delimited)
+    {
+        return identifier;
+    }
+```
+
+#### Example Queries
+
+#### MySQL Table Handling
+
+```
+presto> show schemas from mysql;
+       Schema       
+--------------------
+ Test               
+ TestDb             
+ information_schema 
+ performance_schema 
+ sys                
+ testdb             
+(6 rows)
+
+presto> show tables from mysql.TestDb;
+   Table   
+-----------
+ Test      
+ TestTable 
+ testtable 
```

Comment:
Sounds like, we are making presto case sensitive while remaining backwards compatible. To me the term "Mixed case" across the doc sounds less specific, not sure what others think, but `case sensitive` is more widely used and precise.